docs/content/docs/custom-resource/job-management.md

```diff
@@ -98,8 +98,8 @@ The `upgradeMode` setting controls both the stop and restore mechanisms as detai
 The three  upgrade modes are intended to support different scenarios:
 
  1. **stateless**: Stateless application upgrades from empty state
- 2. **last-state**: Quick upgrades in any application state (even for failing jobs), does not require a healthy job as it always uses last checkpoint information. Manual recovery may be necessary if HA metadata is lost.
- 3. **savepoint**: Use savepoint (when possible) for upgrade, providing maximal safety and possibility to serve as backup/fork point.
+ 2. **last-state**: Quick upgrades in any application state (even for failing jobs), does not require a healthy job as it always uses the latest checkpoint information. Manual recovery may be necessary if HA metadata is lost.
+ 3. **savepoint**: Use savepoint for upgrade, providing maximal safety and possibility to serve as backup/fork point. The savepoint will be created during the upgrade process. Note that the Flink job needs to be running to allow the savepoint to get created. If the job is in an unhealthy state, the last checkpoint will be used. If the last checkpoint is not available, the job upgrade will fail.
```

Comment:
This isn't entirely correct either. Fallback to last checkpoint is a configurable option.

Reply:
Yup, forgot to comment it here: https://issues.apache.org/jira/browse/FLINK-29258

Reply:
I didn't know about the configuration option. I opened the issue because it's impossible for users to know what fallback behavior is configured. This should really be a separate upgrade mode.

Reply:
Originally we had no fallback behavior but in almost all cases users would prefer to upgrade the job even if it’s not running for some reason. There is always a good reason for the upgrade if there is a spec change and savepoint is kind of optional. 

For forking jobs the user can trigger manual savepoints and if they really need the savepoint upgrade they can set the config . Although they will be in big trouble if they must take savepoint and the job is failing for any reason :)